src/mrinufft/operators/base.py

```diff
@@ -500,15 +500,15 @@ def smaps(self):
         return self._smaps
 
     @smaps.setter
-    def smaps(self, smaps):
```

Comment:
I think this is quite complex, maybe refactor the dim check in a dedicated method (`self._check_smaps(smaps)`) that you call instead of using the fset attribute

Reply:
Peace xD.